.github/workflows/rust.yml

```diff
@@ -10,13 +10,46 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  build:
-
+  rustfmt:
```

Comment:
I have a question here. Why are the `rustfmt`, `clippy` and `build` separate jobs rather than more steps in the same job? The latter would be considerably faster, right?

Reply:
Good question! GitHub Actions supports running multiple jobs in parallel. So we better take advantage of that. And if we group all steps together into a single job, one failed step will skip the preceding steps. This is not what we want. We want all errors to be shown. Not one at a time.